api/turing/config/config.go

```diff
@@ -100,7 +107,58 @@ type DeploymentConfig struct {
 
 // EnsemblingJobConfig captures the config related to the ensembling batch jobs.
 type EnsemblingJobConfig struct {
-	DefaultEnvironment string `validate:"required"`
+	BatchSize          int          `validate:"required"`
+	MaxRetryCount      int          `validate:"required"`
+	DefaultEnvironment string       `validate:"required"`
+	KanikoConfig       KanikoConfig `validate:"required"`
+	ImageConfig        ImageConfig  `validate:"required"`
+	InjectGojekConfig  bool
+}
+
+// ImageConfig provides the configuration used for the OCI image building.
+type ImageConfig struct {
+	LocalRegistry        string
```

Comment:
I find these attributes names hard to understand, without looking at the example config. Comments explaining how each of them should be set and/or would be used would be helpful.

Reply:
Ah good thing you pointed this out, there's some residual config that should have been removed, like that LocalRegistry.

Ok I'll add in some comments and see how to make it easier to understand.